client/finality-grandpa/src/justification.rs

```diff
@@ -66,23 +66,41 @@ impl<Block: BlockT> GrandpaJustification<Block> {
 			Err(Error::Client(ClientError::BadJustification(msg)))
 		};
 
+		// we pick the precommit for the lowest block as the base that
+		// should serve as the root block for populating ancestry (i.e.
+		// collect all headers from all precommit blocks to the base)
+		let (base_hash, base_number) = match commit
+			.precommits
+			.iter()
+			.map(|signed| &signed.precommit)
+			.min_by_key(|precommit| precommit.target_number)
+			.map(|precommit| (precommit.target_hash.clone(), precommit.target_number))
+		{
+			None => return error(),
```

Comment:
I think assuming that `precommits` is never empty is valid, `finality-grandpa` crate will never create such a commit. I updated the test to create `GrandpaJustification` directly instead of going through the constructor which will perform some validation.